tango/integrations/beaker/executor.py

```diff
@@ -515,6 +515,7 @@ def update_steps_to_run():
                                 # A dependency failed or can't be run, so this step can't be run.
                                 not_run[step_name] = ExecutionMetadata()
                                 steps_to_run.discard(step_name)
+                                steps_left_to_run.discard(step)
```

Comment:
This line fixes the bug. The other additions below are just for the progress updates.